### ISSUE TITLE
System.Runtime.Tests timing out on Arm64

### ISSUE THREAD

**Issue by ericstj** [dotnet/corefx#38176]
Recently I've been seeing the System.Runtime.Tests on Arm64 fail with timeout:
```
/root/helix/work/workitem /root/helix/work/workitem
  Discovering: System.Runtime.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  System.Runtime.Tests (found 5331 of 5372 test cases)
  Starting:    System.Runtime.Tests (parallel test collections = on, max threads = 46)

[EXECUTION TIMED OUT]
Exit Code:-3
Executor timed out after 900 seconds and was killed
```
https://mc.dot.net/#/user/dotnet-bot/pr~2Fdotnet~2Fcorefx~2Frefs~2Fpull~2F38167~2Fmerge/test~2Ffunctional~2Fcli~2Finnerloop~2F/20190603.16/workItem/System.Runtime.Tests/wilogs

We should investigate whether this is a hang or just too much work in a single test assembly.  If the former: fix it, if the latter: partition the test into multiple assemblies or increase the timeout.

/cc @ViktorHofer

**Comment by ViktorHofer** [comment]
I pinged @MattGal offline to find that out.

cc @stephentoub

**Comment by MattGal** [comment]
To make the same comment as on the other thread, some tests use too many resources to be run in 46-threads-at-once mode without major slowdown or resource contention.

**Comment by danmosemsft** [comment]
Can we simply increase the 900 second timeout on ARM64? If ARM64 is X times slower than X64 generally speaking, it makes sense to have a larger timeout. If there's evidence any tests are pathological on ARM64 we should deal with that separately.

**Comment by ericstj** [comment]
That's already done. https://github.com/dotnet/corefx/pull/38343

**Comment by ViktorHofer** [comment]
No the workitem timeout is defined here: https://github.com/dotnet/corefx/blob/master/eng/sendtohelix.proj#L12

**Comment by ViktorHofer** [comment]
I will add it to the PR to unblock the hangs temporarily.

**Comment by danmosemsft** [comment]
Thanks. Maybe for ARM32 as well?

**Comment by ViktorHofer** [comment]
I didn't see timeout on ARM32?

**Comment by danmosemsft** [comment]
should be fixed by https://github.com/dotnet/corefx/pull/38343
